bothan-core/src/store/errors.rs

```diff
@@ -0,0 +1,24 @@
+use bincode::ErrorKind;
+use thiserror::Error;
+
+#[derive(Clone, Debug, Error, PartialEq, Eq)]
+#[error("An error occurred while storing the data: {message}")]
+pub struct Error {
```

Comment:
Just want to make sure, this is your intention not use from feature of this error to combine error to only one struct not enum right?

Reply:
This is intentional as there isn't a handle case at the moment but switching it to an enum later if needed can be done